### PR TITLE
Executors: add multi-queue configuration

### PR DESCRIPTION
**Pull request by sanderginn**
- Closes https://github.com/sourcegraph/sourcegraph/issues/50616
- Coupled with https://github.com/sourcegraph/sourcegraph/pull/52016 (this is a child branch)

This PR:
- Adds an env var `EXECUTOR_QUEUE_NAMES` to the config of an executor, which supports a comma-separated string of queue names to listen to
- Validates that one and only one of `EXECUTOR_QUEUE_NAME` and `EXECUTOR_QUEUE_NAMES` is configured
- Sets the value on the `apiWorkerOptions` and `queueOptions`

The actual client implementation is found in the parent PR.

## TODO
- [x] Update docs
- [ ] Update Terraform modules: see https://github.com/sourcegraph/sourcegraph/issues/52250

## Test plan
- [x] Unit tests
- [ ] Demo (will be added to parent PR)

### CODE REVIEW

File: enterprise/cmd/executor/internal/config/config.go [hunk 1]
```diff
@@ -97,7 +98,8 @@ func NewAppConfig() *Config {
 func (c *Config) Load() {
 	c.FrontendURL = c.Get("EXECUTOR_FRONTEND_URL", "", "The external URL of the sourcegraph instance.")
 	c.FrontendAuthorizationToken = c.Get("EXECUTOR_FRONTEND_PASSWORD", c.defaultFrontendPassword, "The authorization token supplied to the frontend.")
-	c.QueueName = c.Get("EXECUTOR_QUEUE_NAME", "", "The name of the queue to listen to.")
+	c.QueueName = c.GetOptional("EXECUTOR_QUEUE_NAME", "The name of the queue to listen to.")
+	c.QueueNames = c.GetOptional("EXECUTOR_QUEUE_NAMES", "The names of multiple queues to listen to, comma-separated.")
```
Comment (Piszmog) [inline enterprise/cmd/executor/internal/config/config.go:102]: Include this in the docs - `deploy_executors_binary.md`

File: enterprise/cmd/executor/internal/config/config.go [hunk 2]
```diff
@@ -179,8 +181,18 @@ func getKubeConfigPath() string {
 }
 
 func (c *Config) Validate() error {
-	if c.QueueName != "" && c.QueueName != "batches" && c.QueueName != "codeintel" {
+	if c.QueueName == "" && c.QueueNames == "" {
+		c.AddError(errors.New("One of EXECUTOR_QUEUE_NAME and EXECUTOR_QUEUE_NAMES should be set"))
```
Comment (Piszmog) [inline enterprise/cmd/executor/internal/config/config.go:185]:
```suggestion
		c.AddError(errors.New("Neither EXECUTOR_QUEUE_NAME or EXECUTOR_QUEUE_NAMES is set"))
```

Slight change to verbage. `One of` didn't roll off the tongue very well for me

File: enterprise/cmd/executor/internal/config/config.go [hunk 3]
```diff
@@ -179,8 +181,18 @@ func getKubeConfigPath() string {
...
+	} else if c.QueueName != "" && c.QueueNames != "" {
+		c.AddError(errors.New("Only one of EXECUTOR_QUEUE_NAME and EXECUTOR_QUEUE_NAMES should be set"))
```
Comment (Piszmog) [inline enterprise/cmd/executor/internal/config/config.go:187]: Same here, slight change (I don't really like my suggestion, but something to that gist)

```suggestion
		c.AddError(errors.New("Only set either EXECUTOR_QUEUE_NAME or EXECUTOR_QUEUE_NAMES"))
```

File: enterprise/cmd/executor/internal/run/util.go [hunk 4]
```diff
@@ -60,10 +60,11 @@ func newQueueTelemetryOptions(ctx context.Context, runner util.CmdRunner, useFir
 }
 
 func apiWorkerOptions(c *config.Config, queueTelemetryOptions queue.TelemetryOptions) apiworker.Options {
+	queueNames := strings.Split(c.QueueNames, ",")
```
Comment (Piszmog) [inline enterprise/cmd/executor/internal/run/util.go:63]: you can move this down directly in the struct property, e.g.

```golang
QueueNames: strings.Split(c.QueueNames, ",")
```

File: enterprise/cmd/executor/internal/run/util.go [hunk 5]
```diff
@@ -135,10 +136,11 @@ func resourceOptions(c *config.Config) command.ResourceOptions {
 }
 
 func queueOptions(c *config.Config, telemetryOptions queue.TelemetryOptions) queue.Options {
+	queueNames := strings.Split(c.QueueNames, ",")
```
Comment (Piszmog) [inline enterprise/cmd/executor/internal/run/util.go:139]: same

**Review by Piszmog — COMMENTED** [review]
Great work! Some minor nits and an update to the docs.

**Comment by sanderginn** [comment]
Created https://github.com/sourcegraph/sourcegraph/issues/52250 to support multiple queues for Terraform deployments

File: enterprise/cmd/executor/internal/config/config.go (hunk 1 above)
Comment (sanderginn) [reply enterprise/cmd/executor/internal/config/config.go:102]: Updated as well as other executor docs referencing queue env vars

**Comment by sanderginn** [comment]
Build fails on some unrelated stuff, should be OK in the parent branch which merges into main

**Pull request merged**